process/block/metablock.go

```diff
@@ -622,17 +621,23 @@ func (mp *metaProcessor) saveLastNotarizedHeader(header *block.MetaBlock) error
 		tmpLastNotarizedHdrForShard[i] = mp.lastNotarizedHdrForShard(i)
 	}
 
-	for i := 0; i < len(header.ShardInfo); i++ {
-		shardData := header.ShardInfo[i]
-		header, err := process.GetShardHeaderFromPool(shardData.HeaderHash, mp.dataPool.ShardHeaders())
-		if err != nil {
-			return err
+	mp.hdrsForCurrBlock.mutHdrsForBlock.RLock()
```

Comment:
the code is much cleaner if you iterate on metaHeader.ShardInfo.

Reply:
Agree. The same response as the one above

Reply:
we might forget that this refactoring is needed in that Jira task - so add the requirements there, or the comments from this pull request. It is not necessarily good to have all these change requests postponed for another PR, this is an easy change which could be done for this PR.

Reply:
Done in this PR